biz.aQute.remote/src/aQute/remote/agent/Shell.java

```diff
@@ -49,6 +49,7 @@ public synchronized void add(String s) throws IOException {
 			running = true;
 			try {
 				Object result = session.execute(s);
+				this.out.flush();
```

Comment:
What about flush after the println and printStackTrace below? Perhaps the flush call should be at the bottom of the loop before the call to prompt?